### PR TITLE
[INTERNAL] Refactor tests to get rid of chai and chai-fs

### PR DESCRIPTION
**Pull request by flovogt**
`chai-fs` is not compatible with latest version of `chai`. Therefore its easier to get rid of chai completely.

JIRA: CPOUI5FOUNDATION-801

### CODE REVIEW

File: test/lib/adapters/FileSystem_write.js [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
-import chaifs from "chai-fs";
-chai.use(chaifs);
-const assert = chai.assert;
-
 import sinon from "sinon";
 
 import {createAdapter, createResource} from "../../../lib/resourceFactory.js";
 
+function getFileContent(path) {
+	return readFileSync(path, "utf8");
```
Comment (RandomByte) [inline test/lib/adapters/FileSystem_write.js:12]: If it's not too much trouble I would prefer to use the async API via [`node:fs/promises`](https://nodejs.org/api/fs.html#promises-api). Reading files non-blocking might even have an impact on the execution time of the tests

**Review by RandomByte — CHANGES_REQUESTED** [review]

File: test/lib/adapters/FileSystem_write.js (hunk 1 above)
Comment (flovogt) [reply test/lib/adapters/FileSystem_write.js]: Good point. I changed it to use the [`node:fs/promises`](https://nodejs.org/api/fs.html#promises-api). 
We could use Promise.all for reading actual and expected file in parallel. For simplicity I did not use that but I don't mind. Feel free to propose your preferred option.

File: test/lib/adapters/FileSystem_write.js [hunk 2]
```diff
@@ -45,8 +56,8 @@ test("Write resource", async (t) => {
 	// Write resource content to another readerWriter
 
 	await readerWriters.dest.write(resource);
-	t.notThrows(() => {
-		assert.fileEqual(destFsPath, "./test/fixtures/application.a/webapp/index.html");
+	t.notThrows(async () => {
+		await fileEqual(t, destFsPath, "./test/fixtures/application.a/webapp/index.html");
 	});
```
Comment (RandomByte) [inline test/lib/adapters/FileSystem_write.js:61]:
```suggestion
	await t.notThrowsAsync(fileEqual(t, destFsPath, "./test/fixtures/application.a/webapp/index.html"));
```

Should already do the trick

**Review by RandomByte — CHANGES_REQUESTED** [review]

File: test/lib/adapters/FileSystem_write.js (hunk 1 above)
Comment (RandomByte) [reply test/lib/adapters/FileSystem_write.js]: Fine with me!

File: test/lib/adapters/FileSystem_write.js (hunk 2 above)
Comment (flovogt) [reply test/lib/adapters/FileSystem_write.js:61]: doing mass updates without looking at the actual code is not the best idea :-D

**Review by RandomByte — APPROVED** [review]

**Pull request merged**